packages/components/pie-cookie-banner/src/index.ts

```diff
@@ -86,6 +99,65 @@ export class PieCookieBanner extends LitElement implements CookieBannerProps {
         this._isModalOpen = true;
     };
 
+    /**
+     * Handles the logic of the toggle switch nodes (preferences).
+     * Clicking the “all” toggle switch should turn on all preferences.
+     * When the “all” toggle is checked, and one of the other preferences is clicked,
+     * then the “all” toggle should be unchecked.
+     */
+    private _handlePreferencesToggled = (e: CustomEvent) : void => {
+        const { id } = e?.currentTarget as HTMLInputElement;
+        const turnAllNode = [...this._preferencesNodes].find((node) => node.id === 'all') as PieToggleSwitch;
+
+        if (id === turnAllNode.id) {
+            const isChecked = e.detail;
+            this._preferencesNodes.forEach((node) => {
+                node.isChecked = node.isDisabled ? node.isChecked : isChecked;
+            });
+        } else {
+            turnAllNode.isChecked ||= false;
+        }
+    };
+
+    /**
+     * Renders the content of the preference item.
+     * @private
+     */
+    private renderPreference ({
+        id, title, description, isChecked, isDisabled, hasDivider,
+    }: Preference): TemplateResult {
+        return html`
+            <div class="c-cookieBanner-preferences-item">
```

Comment:
The naming here implies that there is a `c-cookieBanner-preferences` class which `-item` and `-preferences` are children of.

I'd rename this class to be `c-cookieBanner-preferences`:

```suggestion
            <div class="c-cookieBanner-preferences">
```

Reply:
That's fair, as I initially had a wrapper called “c-cookie-banner-preferences” for all preferences, but since we don't have that anymore and we are applying the class to one preference row, should not this be “c-cookie banner-preference” instead of plural?